### PR TITLE
Validation Server Integration

### PR DESCRIPTION
**Pull request by ole-lilienthal**

### CODE REVIEW

File: services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java [hunk 1]
```diff
@@ -19,57 +19,97 @@
...
+    String json = "{ \"tan\": \"" + tan + "\" }";
+    HttpEntity<String> entity = new HttpEntity<>(json, requestHeader);
 
-  private boolean validateTeleTan(String teleTan) {
-    // FIXME Add implementation
-    return true;
+    try {
+      ResponseEntity<String> response = restTemplate.postForEntity(verificationServiceUrl, entity, String.class);
+      return response.getStatusCode().is2xxSuccessful();
+    } catch (HttpClientErrorException e) {
```
Comment (michael-burwig) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java:109]: can be further narrowed down to ``HttpClientErrorException.NotFound``

File: services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java [hunk 2]
```diff
@@ -19,57 +19,97 @@
...
-  private boolean validateTan(String tan) {
-    // FIXME Add implementation
-    return true;
-  }
+  /**
+   * Queries the configured verification service to validate the provided TAN.
+   *
+   * @param tan Submission Authorization TAN
+   * @return {@literal true} if verification service is able to verify the provided TAN, {@literal false} otherwise
+   * @throws RestClientException if http status code is neither 2xx nor 4xx
```
Comment (michael-burwig) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java:100]: can be narrowed to 404

File: services/submission/src/main/resources/application.yaml [hunk 3]
```diff
@@ -14,6 +14,10 @@ services:
     retention-days: 14
     payload:
       max-number-of-keys: 14
+    verification:
+      base-url: ${POSTGRESQL_SERVICE_HOST:http://localhost:8004}
+      path: /version/v1/tan/verify
+
```
Comment (michael-burwig) [inline services/submission/src/main/resources/application.yaml:20]: unneccessary new-line

File: services/submission/src/test/java/app/coronawarn/server/services/submission/verification/TanVerifierTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.client.ExpectedCount;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.HttpServerErrorException.InternalServerError;
+
+@EnableConfigurationProperties(value = SubmissionServiceConfig.class)
+@ExtendWith(SpringExtension.class)
```
Comment (michael-burwig) [inline services/submission/src/test/java/app/coronawarn/server/services/submission/verification/TanVerifierTest.java:51]: redundant due to ``@RestClientTest``

File: services/submission/src/main/resources/application.yaml [hunk 5]
```diff
@@ -14,6 +14,10 @@ services:
     retention-days: 14
     payload:
       max-number-of-keys: 14
+    verification:
+      base-url: ${POSTGRESQL_SERVICE_HOST:http://localhost:8004}
```
Comment (christian-kirschnick) [inline services/submission/src/main/resources/application.yaml:18]: `POSTGRESQL_SERVICE_HOST` looks weird here - probably meant somehting else

File: services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java [hunk 6]
```diff
@@ -19,57 +19,97 @@
...
-        return validateTeleTan(tanAuthorization.getKey());
-      default:
-        throw new UnsupportedOperationException("Authorization Type not supported: " + authType);
+  /**
+   * Verifies if the provided TAN can be parsed as a UUID.
+   *
+   * @param tan Submission Authorization TAN
+   * @return {@literal true} if tan can be parsed as a UUID, {@literal false} otherwise
+   */
+  private boolean checkTanSyntax(String tan) {
```
Comment (michael-burwig) [inline services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java:86]: We could consider tackling this with a ``@pattern`` annotation on the controller method.
see ``@RequestHeader(value = "cwa-authorization") @Pattern(regexp = MY_FAVORITE_URI_REGEX) String tan``

Comment (michael-burwig) [reply services/submission/src/main/java/app/coronawarn/server/services/submission/verification/TanVerifier.java:86]: you mean a meticulously hand-crafted regex

**Pull request merged**